packages/edit-navigation/src/components/name-editor/index.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useEffect, useRef, useState } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import { InspectorControls } from '@wordpress/block-editor';
+import { PanelBody, BaseControl } from '@wordpress/components';
+import { useMenuEntity, useNavigationEditorMenu } from '../../hooks';
+import { useInstanceId } from '@wordpress/compose';
+
+export function NameEditor( {
+	setIsMenuNameEditFocused,
+	isMenuNameEditFocused,
+} ) {
+	const { menuName, menuId } = useNavigationEditorMenu();
+	const { editMenuName } = useMenuEntity( menuId );
+	const inputRef = useRef();
+	const [ tmpMenuName, setTmpMenuName ] = useState( menuName );
+	const instanceId = useInstanceId( NameEditor );
+	const id = `components-edit-navigation-name-editor__input-${ instanceId }`;
+	useEffect( () => setTmpMenuName( menuName ), [ menuName ] );
+	useEffect( () => {
+		if ( isMenuNameEditFocused ) inputRef.current.focus();
+	}, [ isMenuNameEditFocused ] );
+	return (
+		<>
+			<InspectorControls>
+				<PanelBody title={ __( 'Menu Settings' ) }>
+					<BaseControl label={ __( 'Name' ) } id={ id }>
+						<div>
+							<input
```

Comment:
There should be a TextControl component that supports all this:
https://wordpress.github.io/gutenberg/?path=/story/components-textcontrol--default

The description can be added as part of the `helpText` prop.